kasa/tests/smart/modules/test_childprotection.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+from kasa import Module
+from kasa.smart.modules import ChildProtection
+from kasa.tests.device_fixtures import parametrize
+
+child_protection = parametrize(
+    "has child protection",
+    component_filter="child_protection",
+    protocol_filter={"SMART.CHILD"},
+)
+
+
+@child_protection
+@pytest.mark.parametrize(
+    ("feature", "prop_name", "type"),
+    [
+        ("child_lock", "enabled", bool),
+    ],
+)
+async def test_features(dev, feature, prop_name, type):
+    """Test that features are registered and work as expected."""
+    protect: ChildProtection = dev.modules[Module.ChildProtection]
+
+    assert protect is not None
+
+    prop = getattr(protect, prop_name)
+    assert isinstance(prop, type)
+
+    feat = protect._device.features[feature]
+    assert feat.value == prop
```

Comment:
I think we should test the setter as well but your call.

Reply:
Iirc that's tested implicitly through the generic feature tester, but there's probably no harm in having an explicit test for the API.

Reply:
I think the setter came up as uncovered but maybe I made a mistake as I’m on my phone.

Reply:
Added tests. Looks like we didn't have a nice way to handle setters that change the main sysinfo, so I added a helper for that.